packages/expo-cli/src/commands/client/index.js

```diff
@@ -42,8 +42,17 @@ export default program => {
       const distributionCert = await selectDistributionCert(context);
       const pushKey = await selectPushKey(context);
 
+      if (pushKey === null) {
+        log(
+          `Push notifications will be disabled until you upload your push credentials. See https://docs.expo.io/versions/latest/guides/adhoc-builds/#push-notifications-arent-working for more details.`
+        );
+      }
+
       // if user is logged in, then we should update credentials
-      const credentialsList = [distributionCert, pushKey];
+      const credentialsList = [distributionCert, pushKey].filter(
+        // https://stackoverflow.com/questions/18808226/why-is-typeof-null-object
+        cred => typeof cred === 'object' && cred !== null
+      );
```

Comment:
Hmmm, `.filter(i => i)` should be enough. 
```
> [{}, null, undefined, {}].filter(i => i)
[ {}, {} ]
```

Reply:
Hey @quinlanj, would you mind changing this `filter`? I think it unnecessarily obfuscates the code.